EzMap.Api/Controllers/PoiController.cs

```diff
@@ -71,4 +65,30 @@ public async Task<IActionResult> Delete(Guid id, [FromServices] IUnitOfWork uow)
 
         return new StatusCodeResult(StatusCodes.Status500InternalServerError);
     }
+
+    [Authorize]
+    [HttpGet("{id:guid}")]
+    public async Task<IActionResult> GetPoiDetails(Guid id, [FromServices] IIdentityService identityService,
+        [FromServices] IUnitOfWork uow)
+    {
+        if (string.IsNullOrEmpty(id.ToString()))
+        {
+            return BadRequest("Please provide a valid id!");
+        }
+
+        var result = await uow.PoiRepository.GetPoiById(identityService.GetUserId(), id);
+
+        return result is not null ? Ok(result) : new StatusCodeResult(StatusCodes.Status500InternalServerError);
+    }
+
+    [Authorize]
+    [HttpGet]
+    public async Task<IActionResult> GetListPoi([FromServices] IUnitOfWork uow,
+        [FromServices] IIdentityService identityService)
+    {
+
+        var result = await uow.PoiRepository.GetListPoiAsync(identityService.GetUserId());
+
+        return result.Count > 0 ? Ok(result) : Ok("Currently, there is no poi!");
```

Comment:
This is not okay, imagine the front end is expecting some array, but they got a string "Currently, there is no poi!".
There will be parsing error in that case (string is not json).
There's 2 ways to solve this:
1. You can return 204 NoContent
2. You can return empty array.

Reply:
Also same for all other places. If the frontend doesn't need the message, the status alone is sufficient (200, 201, 204 ...), generally 200 is acceptable in most cases.